provider/implementations/js/src/index.ts

```diff
@@ -48,10 +48,10 @@ export class EthereumProviderPlugin extends Module<ProviderConfig> {
       if (
         err && err.message &&
         err.message.indexOf("0x2") > -1 &&
-        params.type === "0x02" &&
+        params[0].type === "0x02" &&
```

Comment:
We should make sure this doesn't throw an exception. Something more like this would be safer:
`Array.isArray(params) && params.length > -1 && params[0].type === "0x02"`